tests/integration/values.rs

```diff
@@ -34,16 +45,29 @@ fn it_accepts_template_values_file() {
         .arg("generate")
         .arg("--name")
         .arg("foobar-project")
+        .arg("--define")
+        .arg("MY_VALUE4=def-value")
+        .arg("--git")
+        .arg(template.path())
         .arg("--template-values-file")
         .arg(template.path().join("my-values.toml"))
-        .arg(template.path())
         .current_dir(&dir.path())
+        .env(
+            "CARGO_GENERATE_TEMPLATE_VALUES_FILE",
+            template.path().join("my-env-values.toml"),
+        )
+        .env("CARGO_GENERATE_VALUE_MY_VALUE2", "env-def-value")
+        .env("CARGO_GENERATE_VALUE_MY_VALUE3", "env-def-value")
```

Comment:
the test illustrates it all, but I'd rather split it up for 2 reasons:
1. if it fails it would be very hard to spot where in the whole mechanics it failed
2. the mental load to oversee the entire mechanics in this test is not quite small